data/actions/scripts/other/potions.lua

```diff
@@ -33,126 +77,81 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
-			return false
-		end
-
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(emptyPot, 1)
-	elseif itemId == healthPot then
-		if not doTargetCombatHealth(0, target, COMBAT_HEALING, 125, 175, CONST_ME_MAGIC_BLUE) then
-			return false
-		end
-
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(emptyPot, 1)
-	elseif itemId == manaPot then
-		if not doTargetCombatMana(0, target, 75, 125, CONST_ME_MAGIC_BLUE) then
-			return false
-		end
-
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(emptyPot, 1)
-	elseif itemId == strongHealthPot then
-		if (not isInArray({3, 4, 7, 8}, target:getVocation():getId()) or target:getLevel() < 50) and not getPlayerFlagValue(player, PlayerFlag_IgnoreSpellCheck) then
-			player:say("This potion can only be consumed by paladins and knights of level 50 or higher.", TALKTYPE_MONSTER_SAY)
-			return true
-		end
-
-		if not doTargetCombatHealth(0, target, COMBAT_HEALING, 250, 350, CONST_ME_MAGIC_BLUE) then
-			return false
-		end
-
-		player:addCondition(exhaust)
-		target:say("Aaaah...", TALKTYPE_MONSTER_SAY)
-		item:remove(1)
-		player:addItem(strongEmptyPot, 1)
-	elseif itemId == strongManaPot then
-		if (not isInArray({1, 2, 3, 5, 6, 7}, target:getVocation():getId()) or target:getLevel() < 50) and not getPlayerFlagValue(player, PlayerFlag_IgnoreSpellCheck) then
-			player:say("This potion can only be consumed by sorcerers, druids and paladins of level 50 or higher.", TALKTYPE_MONSTER_SAY)
-			return true
-		end
-
-		if not doTargetCombatMana(0, target, 115, 185, CONST_ME_MAGIC_BLUE) then
-			return false
-		end
-
+	local stats = potions[item:getId()]
+	if stats then
+		if ( stats.vocations or stats.minLevel ) and not player:getGroup():getAccess() then
+			if not stats.minLevel then
+				stats.minLevel = 1
+			end
+			if not stats.vocations then
+				stats.vocations = {}
+			end
+			
+			local voclist, mvoc = {}, target:getVocation()
+			while mvoc do
+				table.insert(voclist, mvoc:getId())
+				mvoc = mvoc:getDemotion()
+			end
+			-- Checking vocation, level
+			local isdifferentvocation = true
+			for k, v in ipairs(stats.vocations) do
+				if isInArray(v, voclist) then
+					isdifferentvocation = false
+					break
+				end
+			end
+			voclist = nil
+			if isdifferentvocation or target:getLevel() < stats.minLevel then
+				local str, vocations = "", stats.vocations
+				if #vocations > 0 then
+					for i = 1, #vocations do
+						local voc = Vocation(vocations[i])
+						if not voc:getDemotion() then
+							if i == #vocations then
+								str = str .. (#vocations ~= 1 and " and " or "") .. voc:getName():lower() .. "s"
+							else
+								str = str .. (i ~= 1 and ", " or "") .. voc:getName():lower() .. "s"
+							end
+						end
+					end
+				end
+				vocations = nil
+				if str == "" then
+					str = "all vocations"
```

Comment:
The default value for all vocations is usually "players" and not use "only" (e.g. "This potion can be consumed by players of level 50 or higher."